.github/workflows/pre-commit.yml

```
@@ -6,6 +6,9 @@ on:
   push:
     branches: [main]
 
+permissions:
+  contents: read
+
```

These permissions can go down into the context of the job, and then this can simply be:


```suggestion
permissions: {}
```

so that the permissions are more granularly divided/scoped.